[pango] Stop using G_CONST_RETURN
Tomas Bzatek tbzatek at fedoraproject.org
Thu Jun 16 08:27:57 UTC 2011

    Previous message: [Bug 713126] Review Request: harfbuzz - Text shaping library
    Next message: rpmguard: PASSED; 0 warnings for pango-1.28.4-2.fc16
    Messages sorted by: [ date ] [ thread ] [ subject ] [ author ]

commit eededf8c4ac865d8b35bb87588466c648bda9b05
Author: Tomas Bzatek <tbzatek at redhat.com>
Date:   Thu Jun 16 10:27:41 2011 +0200

    Stop using G_CONST_RETURN

--- modules/hebrew/hebrew-shaper.c.const	2009-11-23 07:19:48.000000000 +0100
+++ modules/hebrew/hebrew-shaper.c	2011-06-16 10:22:58.000000000 +0200
@@ -167,7 +167,7 @@ static const gboolean compose_table[4][4
 #define	is_composible(cur_wc, nxt_wc)	(compose_table[char_type_table[ucs2iso8859_8 (cur_wc)]]\
 						      [char_type_table[ucs2iso8859_8 (nxt_wc)]])
 
-G_CONST_RETURN char *
+const char *
 hebrew_shaper_get_next_cluster(const char      *text,
 			       gint		length,
 			       gunichar        *cluster,
--- pango/fonts.c.const	2011-04-04 22:21:10.000000000 +0200
+++ pango/fonts.c	2011-06-16 10:22:58.000000000 +0200
@@ -165,7 +165,7 @@ pango_font_description_set_family_static
  *               %NULL if not previously set.  This has the same life-time
  *               as the font description itself and should not be freed.
  **/
-G_CONST_RETURN char *
+const char *
 pango_font_description_get_family (const PangoFontDescription *desc)
 {
   g_return_val_if_fail (desc != NULL, NULL);
@@ -1927,7 +1927,7 @@ pango_font_family_init (PangoFontFamily
  * Return value: the name of the family. This string is owned
  *   by the family object and must not be modified or freed.
  **/
-G_CONST_RETURN char *
+const char *
 pango_font_family_get_name (PangoFontFamily  *family)
 {
   g_return_val_if_fail (PANGO_IS_FONT_FAMILY (family), NULL);
@@ -2060,7 +2060,7 @@ pango_font_face_is_synthesized (PangoFon
  * Return value: the face name for the face. This string is
  *   owned by the face object and must not be modified or freed.
  **/
-G_CONST_RETURN char *
+const char *
 pango_font_face_get_face_name (PangoFontFace *face)
 {
   g_return_val_if_fail (PANGO_IS_FONT_FACE (face), NULL);
--- pango/pango-attributes.c.const	2011-04-04 22:22:50.000000000 +0200
+++ pango/pango-attributes.c	2011-06-16 10:22:58.000000000 +0200
@@ -97,7 +97,7 @@ pango_attr_type_register (const gchar *n
  *
  * Since: 1.22
  **/
-G_CONST_RETURN char *
+const char *
 pango_attr_type_get_name (PangoAttrType type)
 {
   const char *result = NULL;
--- pango/pango-attributes.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-attributes.h	2011-06-16 10:22:58.000000000 +0200
@@ -180,7 +180,7 @@ struct _PangoAttrFontDesc
 };
 
 PangoAttrType         pango_attr_type_register (const gchar        *name);
-G_CONST_RETURN char * pango_attr_type_get_name (PangoAttrType       type) G_GNUC_CONST;
+const char *          pango_attr_type_get_name (PangoAttrType       type) G_GNUC_CONST;
 
 void             pango_attribute_init        (PangoAttribute       *attr,
 					      const PangoAttrClass *klass);
--- pango/pangocairo-context.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pangocairo-context.c	2011-06-16 10:22:58.000000000 +0200
@@ -260,7 +260,7 @@ pango_cairo_context_set_font_options (Pa
  *
  * Since: 1.10
  **/
-G_CONST_RETURN cairo_font_options_t *
+const cairo_font_options_t *
 pango_cairo_context_get_font_options (PangoContext *context)
 {
   PangoCairoContextInfo *info;
@@ -286,7 +286,7 @@ pango_cairo_context_get_font_options (Pa
  * Return value: the combined set of font options. This value is owned
  * by the context and must not be modified or freed.
  **/
-G_CONST_RETURN cairo_font_options_t *
+const cairo_font_options_t *
 _pango_cairo_context_get_merged_font_options (PangoContext *context)
 {
   PangoCairoContextInfo *info = get_context_info (context, TRUE);
--- pango/pango-context.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-context.c	2011-06-16 10:22:58.000000000 +0200
@@ -188,7 +188,7 @@ pango_context_set_matrix (PangoContext
  *
  * Since: 1.6
  **/
-G_CONST_RETURN PangoMatrix *
+const PangoMatrix *
 pango_context_get_matrix (PangoContext *context)
 {
   g_return_val_if_fail (PANGO_IS_CONTEXT (context), NULL);
--- pango/pango-context.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-context.h	2011-06-16 10:22:58.000000000 +0200
@@ -84,9 +84,9 @@ void                      pango_context_
 							      PangoGravityHint            hint);
 PangoGravityHint          pango_context_get_gravity_hint     (PangoContext               *context);
 
-void                        pango_context_set_matrix (PangoContext      *context,
-						      const PangoMatrix *matrix);
-G_CONST_RETURN PangoMatrix *pango_context_get_matrix (PangoContext      *context);
+void                      pango_context_set_matrix           (PangoContext      *context,
+						              const PangoMatrix *matrix);
+const PangoMatrix *       pango_context_get_matrix           (PangoContext      *context);
 
 /* Break a string of Unicode characters into segments with
  * consistent shaping/language engine and bidrectional level.
--- pango/pango-font.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-font.h	2011-06-16 10:22:58.000000000 +0200
@@ -117,7 +117,7 @@ void                 pango_font_descript
 							       const char           *family);
 void                 pango_font_description_set_family_static (PangoFontDescription *desc,
 							       const char           *family);
-G_CONST_RETURN char *pango_font_description_get_family        (const PangoFontDescription *desc) G_GNUC_PURE;
+const char          *pango_font_description_get_family        (const PangoFontDescription *desc) G_GNUC_PURE;
 void                 pango_font_description_set_style         (PangoFontDescription *desc,
 							       PangoStyle            style);
 PangoStyle           pango_font_description_get_style         (const PangoFontDescription *desc) G_GNUC_PURE;
@@ -212,7 +212,7 @@ GType      pango_font_family_get_type
 void                 pango_font_family_list_faces (PangoFontFamily  *family,
 						   PangoFontFace  ***faces,
 						   int              *n_faces);
-G_CONST_RETURN char *pango_font_family_get_name   (PangoFontFamily  *family) G_GNUC_PURE;
+const char *pango_font_family_get_name   (PangoFontFamily  *family) G_GNUC_PURE;
 gboolean   pango_font_family_is_monospace         (PangoFontFamily  *family) G_GNUC_PURE;
 
 #ifdef PANGO_ENABLE_BACKEND
@@ -261,7 +261,7 @@ struct _PangoFontFamilyClass
 GType      pango_font_face_get_type       (void) G_GNUC_CONST;
 
 PangoFontDescription *pango_font_face_describe       (PangoFontFace  *face);
-G_CONST_RETURN char  *pango_font_face_get_face_name  (PangoFontFace  *face) G_GNUC_PURE;
+const char           *pango_font_face_get_face_name  (PangoFontFace  *face) G_GNUC_PURE;
 void                  pango_font_face_list_sizes     (PangoFontFace  *face,
 						      int           **sizes,
 						      int            *n_sizes);
--- pango/pango-fontmap.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-fontmap.c	2011-06-16 10:22:58.000000000 +0200
@@ -290,7 +290,7 @@ pango_font_map_real_load_fontset (PangoF
  *
  * Since: 1.4
  **/
-G_CONST_RETURN char *
+const char *
 pango_font_map_get_shape_engine_type (PangoFontMap *fontmap)
 {
   g_return_val_if_fail (PANGO_IS_FONT_MAP (fontmap), NULL);
--- pango/pangoft2.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pangoft2.c	2011-06-16 10:22:58.000000000 +0200
@@ -520,7 +520,7 @@ ft_error_compare (const void *pkey,
   return ((ft_error_description *) pkey)->code - ((ft_error_description *) pbase)->code;
 }
 
-G_CONST_RETURN char *
+const char *
 _pango_ft2_ft_strerror (FT_Error error)
 {
 #undef __FTERRORS_H__
--- pango/pango-language.c.const	2011-04-04 22:21:10.000000000 +0200
+++ pango/pango-language.c	2011-06-16 10:22:58.000000000 +0200
@@ -319,7 +319,7 @@ pango_language_from_string (const char *
  * Returns: a string representing the language tag.  This is owned by
  *          Pango and should not be freed.
  */
-G_CONST_RETURN char *
+const char *
 (pango_language_to_string) (PangoLanguage *language)
 {
   return pango_language_to_string (language);
@@ -531,7 +531,7 @@ static const LangInfo lang_texts[] = {
  * Return value: the sample string. This value is owned by Pango
  *   and should not be freed.
  **/
-G_CONST_RETURN char *
+const char *
 pango_language_get_sample_string (PangoLanguage *language)
 {
   const LangInfo *lang_info;
@@ -592,7 +592,7 @@ pango_language_get_sample_string (PangoL
  
  * Since: 1.22
  **/
-G_CONST_RETURN PangoScript *
+const PangoScript *
 pango_language_get_scripts (PangoLanguage *language,
 			    int           *num_scripts)
 {
--- pango/pango-language.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-language.h	2011-06-16 10:22:58.000000000 +0200
@@ -34,11 +34,11 @@ typedef struct _PangoLanguage PangoLangu
 GType          pango_language_get_type    (void) G_GNUC_CONST;
 PangoLanguage *pango_language_from_string (const char *language);
 
-G_CONST_RETURN char *pango_language_to_string   (PangoLanguage *language) G_GNUC_CONST;
+const char    *pango_language_to_string   (PangoLanguage *language) G_GNUC_CONST;
 /* For back compat.  Will have to keep indefinitely. */
 #define pango_language_to_string(language) ((const char *)language)
 
-G_CONST_RETURN char *pango_language_get_sample_string (PangoLanguage *language) G_GNUC_CONST;
+const char    *pango_language_get_sample_string (PangoLanguage *language) G_GNUC_CONST;
 PangoLanguage *pango_language_get_default (void) G_GNUC_CONST;
 
 gboolean      pango_language_matches  (PangoLanguage *language,
@@ -48,7 +48,7 @@ gboolean      pango_language_matches  (P
 
 gboolean		    pango_language_includes_script (PangoLanguage *language,
 							    PangoScript    script) G_GNUC_PURE;
-G_CONST_RETURN PangoScript *pango_language_get_scripts	   (PangoLanguage *language,
+const PangoScript          *pango_language_get_scripts	   (PangoLanguage *language,
 							    int           *num_scripts);
 
 G_END_DECLS
--- pango/pango-layout.c.const	2011-04-04 22:24:17.000000000 +0200
+++ pango/pango-layout.c	2011-06-16 10:22:58.000000000 +0200
@@ -677,7 +677,7 @@ pango_layout_set_font_description (Pango
  *
  * Since: 1.8
  **/
-G_CONST_RETURN PangoFontDescription *
+const PangoFontDescription *
 pango_layout_get_font_description (PangoLayout *layout)
 {
   g_return_val_if_fail (PANGO_IS_LAYOUT (layout), NULL);
@@ -1087,7 +1087,7 @@ pango_layout_set_text (PangoLayout *layo
  *
  * Return value: the text in the @layout.
  **/
-G_CONST_RETURN char*
+const char*
 pango_layout_get_text (PangoLayout *layout)
 {
   g_return_val_if_fail (PANGO_IS_LAYOUT (layout), NULL);
--- pango/pango-layout.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-layout.h	2011-06-16 10:22:58.000000000 +0200
@@ -116,7 +116,7 @@ void           pango_layout_set_markup_w
 void           pango_layout_set_font_description (PangoLayout                *layout,
 						  const PangoFontDescription *desc);
 
-G_CONST_RETURN PangoFontDescription *pango_layout_get_font_description (PangoLayout *layout);
+const PangoFontDescription *pango_layout_get_font_description (PangoLayout *layout);
 
 void           pango_layout_set_width            (PangoLayout                *layout,
 						  int                         width);
--- pango/pango-ot-ruleset.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-ot-ruleset.c	2011-06-16 10:22:58.000000000 +0200
@@ -107,7 +107,7 @@ pango_ot_ruleset_finalize (GObject *obje
  *
  * Since: 1.18
  **/
-G_CONST_RETURN PangoOTRuleset *
+const PangoOTRuleset *
 pango_ot_ruleset_get_for_description (PangoOTInfo                     *info,
 				      const PangoOTRulesetDescription *desc)
 {
--- pango/pango-renderer.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-renderer.c	2011-06-16 10:22:58.000000000 +0200
@@ -1352,7 +1352,7 @@ pango_renderer_set_matrix (PangoRenderer
  *
  * Since: 1.8
  **/
-G_CONST_RETURN PangoMatrix *
+const PangoMatrix *
 pango_renderer_get_matrix (PangoRenderer *renderer)
 {
   g_return_val_if_fail (PANGO_IS_RENDERER (renderer), NULL);
--- pango/pango-renderer.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-renderer.h	2011-06-16 10:22:58.000000000 +0200
@@ -249,7 +249,7 @@ PangoColor *pango_renderer_get_color (Pa
 
 void                        pango_renderer_set_matrix (PangoRenderer     *renderer,
 						       const PangoMatrix *matrix);
-G_CONST_RETURN PangoMatrix *pango_renderer_get_matrix (PangoRenderer     *renderer);
+const PangoMatrix          *pango_renderer_get_matrix (PangoRenderer     *renderer);
 
 PangoLayout     *pango_renderer_get_layout      (PangoRenderer     *renderer);
 PangoLayoutLine *pango_renderer_get_layout_line (PangoRenderer     *renderer);
--- pango/pango-script.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-script.c	2011-06-16 10:22:58.000000000 +0200
@@ -164,10 +164,10 @@ pango_script_iter_free (PangoScriptIter
  * Since: 1.4
  **/
 void
-pango_script_iter_get_range (PangoScriptIter      *iter,
-			     G_CONST_RETURN char **start,
-			     G_CONST_RETURN char **end,
-			     PangoScript          *script)
+pango_script_iter_get_range (PangoScriptIter  *iter,
+                             const char      **start,
+                             const char      **end,
+                             PangoScript      *script)
 {
   if (start)
     *start = iter->script_start;
--- pango/pango-script.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-script.h	2011-06-16 10:22:58.000000000 +0200
@@ -128,12 +128,12 @@ PangoScript pango_script_for_unichar
 
 PangoScriptIter *pango_script_iter_new       (const char          *text,
 					      int                  length);
-void             pango_script_iter_get_range (PangoScriptIter      *iter,
-					      G_CONST_RETURN char **start,
-					      G_CONST_RETURN char **end,
-					      PangoScript          *script);
-gboolean         pango_script_iter_next      (PangoScriptIter      *iter);
-void             pango_script_iter_free      (PangoScriptIter      *iter);
+void             pango_script_iter_get_range (PangoScriptIter     *iter,
+                                              const char         **start,
+                                              const char         **end,
+                                              PangoScript         *script);
+gboolean         pango_script_iter_next      (PangoScriptIter     *iter);
+void             pango_script_iter_free      (PangoScriptIter     *iter);
 
 #include <pango/pango-language.h>
 
--- pango/pango-utils.c.const	2011-04-04 22:21:10.000000000 +0200
+++ pango/pango-utils.c	2011-06-16 10:22:58.000000000 +0200
@@ -689,7 +689,7 @@ DllMain (HINSTANCE hinstDLL,
  * Return value: the Pango sysconf directory. The returned string should
  * not be freed.
  */
-G_CONST_RETURN char *
+const char *
 pango_get_sysconf_subdirectory (void)
 {
 #ifdef G_OS_WIN32
@@ -718,7 +718,7 @@ pango_get_sysconf_subdirectory (void)
  * Return value: the Pango lib directory. The returned string should
  * not be freed.
  */
-G_CONST_RETURN char *
+const char *
 pango_get_lib_subdirectory (void)
 {
 #ifdef G_OS_WIN32
--- pango/pango-utils.h.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pango-utils.h	2011-06-16 10:22:58.000000000 +0200
@@ -82,13 +82,13 @@ gboolean pango_parse_stretch (const char
  * stored in the registry). The returned string should not be
  * g_free'd.
  */
-G_CONST_RETURN char *   pango_get_sysconf_subdirectory (void) G_GNUC_PURE;
+const char *   pango_get_sysconf_subdirectory (void) G_GNUC_PURE;
 
 /* Ditto for LIBDIR/pango. On Win32, use the same Pango
  * installation directory. This returned string should not be
  * g_free'd either.
  */
-G_CONST_RETURN char *   pango_get_lib_subdirectory (void) G_GNUC_PURE;
+const char *   pango_get_lib_subdirectory (void) G_GNUC_PURE;
 
 #endif /* PANGO_ENABLE_BACKEND */
 
@@ -131,12 +131,12 @@ gboolean pango_is_zero_width (gunichar c
 int pango_version (void) G_GNUC_CONST;
 
 /* Return run-time Pango version as an string */
-G_CONST_RETURN char * pango_version_string (void) G_GNUC_CONST;
+const char * pango_version_string (void) G_GNUC_CONST;
 
 /* Check that run-time Pango is as new as required */
-G_CONST_RETURN char * pango_version_check (int required_major,
-					   int required_minor,
-					   int required_micro) G_GNUC_CONST;
+const char * pango_version_check (int required_major,
+                                  int required_minor,
+                                  int required_micro) G_GNUC_CONST;
 
 G_END_DECLS
 
--- pango/pangox-fontmap.c.const	2010-05-04 17:50:40.000000000 +0200
+++ pango/pangox-fontmap.c	2011-06-16 10:22:58.000000000 +0200
@@ -1441,7 +1441,7 @@ pango_x_fontmap_atom_from_name (PangoFon
 }
 
 
-G_CONST_RETURN char *
+const char *
 pango_x_fontmap_name_from_atom (PangoFontMap *fontmap,
 				Atom          atom)
 {
@@ -1645,7 +1645,7 @@ pango_x_family_list_faces (PangoFontFami
     }
 }
 
-static G_CONST_RETURN char *
+static const char *
 pango_x_family_get_name (PangoFontFamily  *family)
 {
   PangoXFamily *xfamily = PANGO_X_FAMILY (family);
